### PR TITLE
Second batch of React error fixes

### PR DESCRIPTION
**Pull request by martinpitt**
- [x] builds on top of #1270

[pixel diff review](https://github.com/cockpit-project/pixel-test-reference/compare/258785f9457772525da2142dbfc0c1db490b9c86..85f5d7aa3d629c80af2cdf8ac18fa1330c6b018f) for storage detection race fix

### CODE REVIEW

File: src/components/common/needsShutdown.jsx [hunk 1]
```diff
@@ -159,7 +159,10 @@ export const VmNeedsShutdown = ({ vm }) => {
             position="bottom"
             hasAutoWidth
             bodyContent={body}>
-            <Label className="resource-state-text" href="javascript:null" color="blue" id={`vm-${vm.name}-needs-shutdown`} icon={<PendingIcon />}>{_("Changes pending")}</Label>
+            <Label className="resource-state-text" color="blue" id={`vm-${vm.name}-needs-shutdown`}
+                   icon={<PendingIcon />} onClick={() => null}>
```
Comment (cockpituous) [inline src/components/common/needsShutdown.jsx:163]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1274-20231027-124307-1f1884f1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline src/components/common/needsShutdown.jsx:163]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1274-20231027-124301-1f1884f1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

File: src/components/create-vm-dialog/createVmDialog.jsx [hunk 2]
```diff
@@ -947,19 +947,15 @@ const StorageRow = ({ connectionName, allowNoDisk, storageSize, storageSizeUnit,
         <SelectOption key="NewVolumeRAW" value="NewVolumeRAW">{_("Create new raw volume")}</SelectOption>
     ];
     if (allowNoDisk) {
-        StorageSelectOptions.push(<>
-            <Divider key="dividerNoStorage" />
-            <SelectOption value="NoStorage" key="NoStorage">{_("No storage")}</SelectOption>
-        </>);
+        StorageSelectOptions.push(<Divider key="dividerNoStorage" />);
+        StorageSelectOptions.push(<SelectOption value="NoStorage" key="NoStorage">{_("No storage")}</SelectOption>);
```
Comment (jelly) [inline src/components/create-vm-dialog/createVmDialog.jsx:951]: The value is seen in the screen shot that seems wrong.

**Review by jelly — CHANGES_REQUESTED** [review]

File: src/components/create-vm-dialog/createVmDialog.jsx (hunk 2 above)
Comment (martinpitt) [reply src/components/create-vm-dialog/createVmDialog.jsx:951]: Hm, I don't understand how that could have changed, nor can I reproduce it locally, but it did fail again after a retry. :exploding_head:

Comment (martinpitt) [reply src/components/create-vm-dialog/createVmDialog.jsx:951]: I have a plausible theory, added a separate commit to fix the race.

Comment (martinpitt) [reply src/components/create-vm-dialog/createVmDialog.jsx:951]: Haha no, we both [read this wrong](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1274-20231030-135040-a4e4a274-fedora-38/pixeldiff.html#TestMachinesCreate-testCreateBasicValidation-auto) -- it's the *current main* screenshot which is bogus! That's the one that said "NoStorage".

Comment (jelly) [reply src/components/create-vm-dialog/createVmDialog.jsx:951]: Woops! Nice, we are fixing real bugs \o/

**Review by jelly — APPROVED** [review]

**Pull request merged**